File: 프로그래머스/0/120869. 외계어 사전/외계어 사전.js
function solution(spell, dic) {
    const equals = (a, b) => a.length === b.length && a.every((x, i) => x === b[i]);
  
    return dic.filter((x) => equals(spell.sort(), x.split("").sort())).length > 0 ? 1 : 2;
}

File: 프로그래머스/0/120869. 외계어 사전/외계어 사전.py
def solution(spell, dic):
    check_list = [0 for r in range(0, len(dic))]

    for i, word in enumerate(dic):
        for s in spell:
            if s not in word:
                check_list[i] = 2
                break
            else:
                check_list[i] = 1

    return min(check_list)